update.php

```diff
@@ -739,12 +788,40 @@ function pre_update_1364()
 
 function pre_update_1365()
 {
-	if (!DBA::e("DELETE FROM `notify-threads` WHERE NOT `notify-id` IN (SELECT `id` FROM `notify`)")) {
+	if (DBStructure::existsTable('notify-threads') &&
+		!DBA::e("DELETE FROM `notify-threads` WHERE NOT `notify-id` IN (SELECT `id` FROM `notify`)")) {
 		return Update::FAILED;
 	}
 
-	if (!DBA::e("DELETE FROM `thread` WHERE NOT `iid` IN (SELECT `id` FROM `item`)")) {
+	if (DBStructure::existsTable('thread') &&
+		!DBA::e("DELETE FROM `thread` WHERE NOT `iid` IN (SELECT `id` FROM `item`)")) {
+		return Update::FAILED;	}
+
+	return Update::SUCCESS;
+}
+
+function pre_update_1367()
+{
+	if (DBA::count('locks') > 0 &&
+		!DBStructure::existsColumn('locks', ['host-id'])) {
+
+		// If the locks table doesn't have a host-id, but values, you have to temporary disable the foreign checks!
+		if (!DBA::e('SET FOREIGN_KEY_CHECKS=0')) {
```

Comment:
@annando do you think does this work? So the dbstructure is able to create the foreign key constraints and in the `update_1367()` function, I set the right values and re-enable the foreign check?

Reply:
I'm unsure about that. I also don't know what happens when the checks are enabled again and there is still a violation. That's why I delete all violations first.